tools/distribution/src/assets/podspec.py

```diff
@@ -0,0 +1,107 @@
+        self.__file_name = file_name
+        self.__path = file_path
+        print(f'   → `{file_name}` found')
+
+    def __repr__(self):
+        return f'[CPPodspec: name = {self.__name}, path = {self.__path}]'
+
+    def validate(self, git_tag: str):
+        """
+        Checks the `.podspec` integrity with given `git_tag`.
+        """
+        print(f'🔎️️ Validating {self} against: {git_tag}')
+
+        # Check if spec `.version` matches the git tag name and `sdk_version`:
+        sdk_version = read_sdk_version()
+        pod_version = self.__read_pod_version()
+        if not (pod_version == git_tag and sdk_version == git_tag):
+            raise Exception(f'`sdk_version` ({sdk_version}), `pod_version` ({pod_version})'
+                            f' and git tag ({git_tag}) do not match')
+        print(f'   → `sdk_version` ({sdk_version}), `pod_version` ({pod_version})'
+              f' and git tag ({git_tag}) do match')
+
+    def publish(self, dry_run: bool):
+        """
+        Publishes the `.podspec` to pods trunk.
+        """
+        print(f'📦 Publishing {self} ({self.__read_pod_version()})')
+
+        # Because some of our pods depend on others and due to https://github.com/CocoaPods/CocoaPods/issues/9497
+        # we need to retry `pod trunk push` until it succeeds. The maximum number of attempts is 100 (arbitrary),
+        # but this is also limited by the CI job timeout.
+        retry_time = 30  # seconds
+        attempt = 0
+        while True or attempt > 100:  # as trunk may not yet
+            attempt += 1
+            try:
+                if attempt == 1:
+                    shell(f'pod repo update', skip=dry_run)
+                    shell(f'pod spec lint --allow-warnings {self.__file_name}', skip=dry_run)
+                    shell(f'pod trunk push --synchronous --allow-warnings {self.__file_name}', skip=dry_run)
+                else:
+                    shell(f'pod repo update --silent', skip=dry_run)
+                    shell(f'pod spec lint  --silent --allow-warnings {self.__file_name}', skip=dry_run)
+                    shell(f'pod trunk push --allow-warnings {self.__file_name}', skip=dry_run)
+
+                if dry_run and random.choice([True, False]):  # to enable testing in `dry_run` mode
+                    retry_time = 1
+                    raise Exception('Running in dry_run mode, simulating `pod` command failure')
+
+                print(f'   → succeeded in {attempt} attempt(s)')
+                break  # break the while loop once all succeed without raising an exception
+
+            except Exception:
+                print(f'   → failed on attempt {attempt} (retrying in {retry_time}s)')
+
+            time.sleep(retry_time)
+
+    def __read_pod_version(self) -> str:
+        """
+        Reads pod version from podspec file.
+        """
+        version_regex = r'^.*\.version.*=.*\"([0-9]\.[0-9]\.[0-9][\-a-z0-9]*)\"'  # e.g. ' s.version   = "1.7.1-alpha1"'
```

Comment:
```suggestion
        version_regex = r'^.*\.version.*=.*\"([0-9]+\.[0-9]+\.[0-9]+[\-a-z0-9]*)\"'  # e.g. ' s.version   = "1.7.1-alpha1"'
```
Otherwise, it will fail for `1.10.0`.